tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/Matrix.java

```diff
@@ -0,0 +1,563 @@
+      return createFromTensor(scope, value);
+    }
+  }
+
+  /**
+   * Creates a rank-5 constant of {@code float} elements.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param data An array containing the values to put into the new constant. The dimensions of the
+   *     new constant will match those of the array.
+   * @return a {@link TFloat32} constant matrix
+   */
+  public static Matrix<TFloat32> create(Scope scope, float[][][][][] data) {
+    try (Tensor<TFloat32> value = TFloat32.tensorOf(StdArrays.shapeOf(data), t -> StdArrays.copyTo(t, data))) {
+      return createFromTensor(scope, value);
+    }
+  }
+
+  /**
+   * Creates a rank-6 constant of {@code float} elements.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param data An array containing the values to put into the new constant. The dimensions of the
+   *     new constant will match those of the array.
+   * @return a {@link TFloat32} constant matrix
+   */
+  public static Matrix<TFloat32> create(Scope scope, float[][][][][][] data) {
+    try (Tensor<TFloat32> value = TFloat32.tensorOf(StdArrays.shapeOf(data), t -> StdArrays.copyTo(t, data))) {
+      return createFromTensor(scope, value);
+    }
+  }
+
+  /**
+   * Creates a rank-2 constant of {@code double} elements.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param data An array containing the values to put into the new constant. The dimensions of the
+   *     new constant will match those of the array.
+   * @return a {@link TFloat64} constant matrix
+   */
+  public static Matrix<TFloat64> create(Scope scope, double[][] data) {
+    try (Tensor<TFloat64> value = TFloat64.tensorOf(StdArrays.shapeOf(data), t -> StdArrays.copyTo(t, data))) {
+      return createFromTensor(scope, value);
+    }
+  }
+
+  /**
+   * Creates a rank-3 constant of {@code double} elements.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param data An array containing the values to put into the new constant. The dimensions of the
+   *     new constant will match those of the array.
+   * @return a {@link TFloat64} constant matrix
+   */
+  public static Matrix<TFloat64> create(Scope scope, double[][][] data) {
+    try (Tensor<TFloat64> value = TFloat64.tensorOf(StdArrays.shapeOf(data), t -> StdArrays.copyTo(t, data))) {
+      return createFromTensor(scope, value);
+    }
+  }
+
+  /**
+   * Creates a rank-4 constant of {@code double} elements.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param data An array containing the values to put into the new constant. The dimensions of the
+   *     new constant will match those of the array.
+   * @return a {@link TFloat64} constant matrix
+   */
+  public static Matrix<TFloat64> create(Scope scope, double[][][][] data) {
```

Comment:
Is a 4-rank object really a matrix? Should these be moved to `tensorConstant` or some other name? Also the javadoc should probably note that these are slower than other ways of creating a 4-rank object.

Reply:
> Is a 4-rank object really a matrix

`tensor` can also be rank-0 or 1, while here it is only used to initialize rank-2+ tensors from standard nd-arrays. I thought `matrix` meant that in some way but while it is pretty common for 2D and 3D, I personally don't know how awkward it sounds in the field for 4D+ data structures. 

We can also all merge back all those endpoints in a single `constant` (or `tensor`?) operator, with an exception for the `vector` initializers taking advantages of the vararg parameter.

> Also the javadoc should probably note that these are slower than other ways of creating a 4-rank object.

Actually, since I removed all reflective operations on them, their performances is comparable to the other methods. In graph mode, the usage of a `constant` is mainly to instantiate a value from some constant values, which should be relatively small. I think what was really killing performances is when large tensors were created for feeding data to a graph using one of the `Tensors.create` methods, like [this one](https://github.com/tensorflow/java/blob/126176668a56179ba5dc7bc5d152fee27efe0127/tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensors.java#L116) (that I plan to remove as well).

Reply:
Just thought of something, I've always found that `tf.constant` is a bit verbose for something that is used so often.

What about `tf.val` instead? It is concise and kind of implies that it is an immutable (constant) value. Then, the `tf.vector` could be renamed maybe to `tf.array` for vararg initialization only and other operators would be removed. @Craigacp?

Reply:
Trying it out, I think I really like it:
```java
  private static Tensor<TFloat32> preprocessImages(ByteNdArray rawImages) {
    Ops tf = Ops.create();

    // Flatten images in a single dimension and normalize their pixels as floats.
    long imageSize = rawImages.get(0).shape().size();
    return tf.math.div(
        tf.reshape(
            tf.dtypes.cast(tf.val(rawImages), TFloat32.DTYPE),
            tf.array(-1L, imageSize)
        ),
        tf.val(255.0f)
    ).asTensor();
  }
```
Both `val` and `array` return a `Constant` object (with the help of the new `@Endpoint` annotation). I think I'll push those changes, the timing is right for improving the whole API.

Reply:
> Just thought of something, I've always found that tf.constant is a bit verbose for something that is used so often.

Agreed; I like the `tf.val` and `tf.array` with vararg initialization.

I think the necessity of a `tf.matrix` is not very high; it'd be ok to just have `tf.constant` or `tf.tensor` methods to construct higher dimensional tensors like you suggest above.

Reply:
In fact, API speaking, I've reverted most of the signatures to what they were, with the following exceptions:

* `tf.constant` has been renamed to `tf.val`
* `tf.array` has been added to support vararg
* `Shape.make` has been renamed to `Shape.of`
* A few other minor things.

So there is no more `tf.matrix`, `tf.vector`, `tf.scalar`, `tf.tensor`....